File: PartyApp/app/src/main/java/com/example/alex/partyapp/di/ApplicationComponent.java
package com.example.alex.partyapp.di;

import com.example.alex.partyapp.MainActivity;
import com.example.alex.partyapp.loading.LoadingFragment;
import com.example.alex.partyapp.login.LoginFragment;
import com.example.alex.partyapp.servers.ServersFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, NetworkModule.class})
public interface ApplicationComponent {
    void inject(LoginFragment fragment);
    void inject(ServersFragment fragment);
    void inject(MainActivity activity);
    void inject(LoadingFragment fragment);
}

File: app/src/main/java/lt/petraslabutis/testio/dagger/ApplicationComponent.kt
package lt.petraslabutis.testio.dagger

import dagger.Component
import lt.petraslabutis.testio.activities.MainActivity
import lt.petraslabutis.testio.fragments.BaseFragment
import lt.petraslabutis.testio.fragments.LoginFragment
import lt.petraslabutis.testio.fragments.ServerListFragment
import javax.inject.Singleton

@Singleton
@Component(modules = [ApiModule::class, StorageModule::class, AppModule::class, ViewModelModule::class])
interface ApplicationComponent {
    fun inject(fragment: BaseFragment)
    fun inject(fragment: LoginFragment)
    fun inject(fragment: ServerListFragment)
    fun inject(activity: MainActivity)
}